sklearn/tests/test_naive_bayes.py

```diff
@@ -53,6 +53,31 @@ def test_gnb():
     # FIXME Remove this test once the more general partial_fit tests are merged
     assert_raises(ValueError, GaussianNB().partial_fit, X, y, classes=[0, 1])
 
+    # Test label mismatch at time of initialization:
+    assert_raises(ValueError, GaussianNB(classes=[0, 1]).partial_fit, X, y)
+    assert_raises(ValueError, GaussianNB(classes=[0, 1]).fit, X, y)
+
+
+def test_gnb_classes_init_partial_fit():
+    # check if setting classes in both init and partial fit results in an error
+    # to be deprecated when partial fit's classes argument removed
+    assert_raises(ValueError,
+                  GaussianNB(classes=[0, 1]).partial_fit,
+                  X, y, classes=[0, 1])
+
+
+def test_gnb_classes_init_partial_fit_same_result():
+    # check if setting classes during initialization or partial_fit gives the
+    # same results
+    # to be deprecated when partial fit's classes argument removed
+    y_pred1 = GaussianNB(classes=[0, 1, 2, 3]).fit(X, y).predict_proba(X)
+    y_pred2 = GaussianNB().partial_fit(X, y,
```

Comment:
you should catch the deprecation warning here (or add a decorator to the test) and check if this is used anywhere else in the code or in examples.